staging/src/k8s.io/client-go/discovery/aggregated_discovery.go

```diff
@@ -111,6 +111,8 @@ func convertAPIGroup(g apidiscovery.APIGroupDiscovery) (
 	return group, gvResources, failedGVs
 }
 
+var emptyKind = metav1.GroupVersionKind{}
```

Comment:
Would changing this file further add complications of different behaviors across different client versions?
Eg: Communicating with the latest server (v1.29) on v1.28.x and v1.28.y client versions would have different behaviors depending on whether x and y contain these backported changes.

Reply:
I don't really see an alternative to tightening the check here if we don't want to include entries that don't provide kind info, and also don't want to return nil kind info from the server to avoid panicking v0.26.0 clients